relayer/pkg/chainlink/ocr2/contract_reader.go

```diff
@@ -20,14 +21,14 @@ type Reader interface {
 var _ Reader = (*contractReader)(nil)
 
 type contractReader struct {
-	address string
+	address caigotypes.Hash
 	reader  OCR2Reader
 	lggr    logger.Logger
 }
 
 func NewContractReader(address string, reader OCR2Reader, lggr logger.Logger) Reader {
 	return &contractReader{
-		address: address,
+		address: caigotypes.HexToHash(address), // TODO: propagate type everywhere
```

Comment:
is the `TODO` resolved?

Reply:
No, I'd prefer the `NewContractReader` and all parent methods to also take a `Hash` (we could alias it as the `starknet.Address` type)